url6da19b5a7a_krakrak_Adventure_Game_tgz-pJ8-adventure_MainJ8/src/items/SilverCompanionsSword.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package items;
+
+import NPC.Kill;
+import NPC.Killable;
+import NPC.Killable1;
+import NPC.Npc;
+import commands.Game;
+import commands.GameState;
+import commands.History;
+import java.util.List;
+import java.util.Random;
+
+/**
+ *
+ * @author Irakiza Krak
+ */
+public class SilverCompanionsSword extends AbstractWeapon {
+    
+    private Random rdm = new Random();
+    private Double removeNbLives;
+    private int nbLivesRemain;
+    
+    public SilverCompanionsSword(){
+        super("Silver Companion's Sword", "Large iron two-handed sword", "si s");
+    }
+
+    @Override
+    public GameState use(Game game) {
+        List <Npc> npcs;
+        npcs = game.getCurrentRoom().getNpcs();
+        Kill killablenpc = null;
+        
+        for(Npc creature : npcs){
+            if(creature instanceof Killable1){
+                killablenpc = (Kill) creature;               
+            }
+        }
+        if(killablenpc != null){
+            if (!History.getHistory().isLoaded()){
+                nbLivesRemain = killablenpc.getNbLives();
+                if(nbLivesRemain > 0 ){
+                    removeNbLives = rdm.nextDouble()*(100/nbLivesRemain) + 13;
+                    killablenpc.setNbLives(nbLivesRemain - removeNbLives.intValue());
+                    System.out.println(killablenpc.getStatement());
+                    if(killablenpc.getStatement().length() > removeNbLives)
+                        killablenpc.setStatement(killablenpc.getStatement().substring(killablenpc.getStatement().length()- removeNbLives.intValue(), killablenpc.getStatement().length()));
+                    System.out.println("The "+killablenpc.getName()+" has "+killablenpc.getNbLives() +" lives \n "
+                        + "You removed "+removeNbLives.intValue() + " lives");
+                    if(killablenpc.getNbLives() <= 0){
+                        System.out.println("FINISH THE  "+killablenpc.getName().toUpperCase() +"!!");  
+                    }
+                }
+                else {
+                    System.out.println("You overcame "+killablenpc.getName());
+                    for(Item reward : killablenpc.getItems()){
+                        game.getCurrentRoom().addItem(reward);
+                        System.out.println("The "+killablenpc.getName()+ " dropped something look around");
+                    }
+                    game.getCurrentRoom().removeNpcs(killablenpc);
+                }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 57:64, 66:73.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.